accepted/2021/User-Experience/Apple-Silicon-UX.md

```diff
@@ -0,0 +1,107 @@
+- The user or the IDE can explicitly select the architecture of the target process using the `arch` command.
+
+The extra search cost could be minimized by passing the result of the previous search when restarting the process. Adding an `--fx-version` option would accomplish this at the cost of changing the users specified argument order.  Using a new environment variable would avoid this unexpected side-effect.
+
+## Apple Silicon requires Signed Binaries
+
+Apple Silicon with Big Sur adds a new security policy which requires signed binaries by default. Unsigned binaries are simply killed without explanation. The user can expect an inscrutable message like `zsh killed: helloWorld` This directly affects our users who expect published binaries to just work.
+
+The developer would need to explicitly sign after publishing for Apple Silicon.  The sigining command is relatively simple `codesign -s <signature> <app-path>`.
+
+### Proposed changes
+
+This is largely TBD. The addition of a message during publish is the most likely design.
+
+## Universal Binaries
+
+As part of the architectural transition, Apple has switched the entire BigSur OS to use universal binaries. This means every file in the OS has both the `arm64` and the `x86_64` code on disk.
+
+Given both architectures, universal binaries are relatively easy to create. The general syntax would be `lipo <binaryArchA> <binaryArchB> -create -output <binaryUniversal>`
+
+### No planned universal binary .NET runtime
+
+At least for .NET 6, general universal binary support is not planned.
+
+General support would require:
+
+- All native binaries and libraries were built as universal binaries.
+- All managed libraries would contain IL which would work correctly on both platforms (no architecture specific `#if`)
+- All managed libraries contain either no `crossgen` native code, or `crossgen` code for both architectures.
+- The runtime would need to be taught how to select the correct architecture `crossgen` code.
+- The managed runtime file format would need to be extended to support multiple `crossgen` architectures.
+
+Given:
+
+- Universal binaries would effectively double the size of the runtime.
+- It requires significant investment
+
+This did not make .NET 6 investment plan.
+
+#### Shared dotnet host
+
+The impact to the shared `dotnet` host is discussed at length above. `dotnet` will be a universal binary.
+
+#### dotnet-sos and the SOS plugin
+
+The `dotnet-sos` tool installs the `SOS` diagnostic plugin into `lldb`. Since `lldb` is a universal binary, it expects its plugins to also be universal binaries. The plan would be to make the Apple Silcon SOS be a universal binary to allow SOS to work on `x86_64` and `arm64` processes.
+
+#### Apple Store requires universal binaries
+
+Publishing a .NET runtime to the Apple store is not a common customer scenario. However it is worth noting that the Apple store is requiring universal binaries.
```

Comment:
I'd remove or rewrite this sense as it's incorrect. Today all Xamarin developers do exactly this.